src/porepy/models/poromechanics.py

```diff
@@ -190,6 +190,16 @@ def initial_condition(self) -> None:
 
         self.update_time_dependent_ad_arrays(initial=True)
 
+    def reset_state_from_file(self) -> None:
+        """Reset states but through a restart from file.
+
+        Add treatment of boundary conditions to the standard reset of states from file.
+
+        """
+        super().reset_state_from_file()
+
+        self.update_time_dependent_ad_arrays(initial=True)
+
```

Comment:
Tightly connected to the reopened discussion in another discussion. This is a demonstration of the need to tailor `reset_state_from_file` depending on the model. In this case, by separating time dependent boundary condition treatment from intial_condition, one could get rid of this case.

Reply:
Who is `super` in this context?

Reply:
Of course my sight is limited. It is SolutionStrategy. One should think this certainly through and consider what happens in case of models of which multiple ones require a tailored reset routine. My current thinking is that resetting in principle mimicks initial_condition, and if initial_condition works out with all it's "super" calls. A similar setup will also work for resetting. After all, resetting requires: setting states, boundary conditions, parameters. Similar to initial_condition.